### DISCUSSION TITLE
Major discrepancy b/w API and opinion search results

### DISCUSSION THREAD

**Discussion by JakeATX** [freelawproject/courtlistener#4458] [Q&A]
I am trying to research on all cases involving trademark precedent (~36K results claimed in the opinion search page). Using the search API, far less than half of that are found. The web page does not show any results beyond page 100, so I am not able to verify anything manually. Any idea which dataset to believe, or how to best access?

**Comment by mlissner** [comment]
Two things: 

1. Try switching to v4 of the API. It should let you do that deep pagination. It will use our new search engine as well, which will give much better results. 

2. If you can share links, that's very helpful.

**Comment by JakeATX** [comment]
This appears to have worked! The Search API # for the query involved is showing the same estimated number of results. I did run into an issue where some of the results appear to be missing IDs (so I cant get matching opinions), but it seems like that is potentially just a data fidelity limitation based on what I've seen in the documentation?

**Reply by mlissner** [reply]
Example please
